OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/BaseBuilderQueueManager.cs

```diff
@@ -368,11 +371,45 @@ ActorInfo ChooseBuildingToBuild(ProductionQueue queue)
 			return null;
 		}
 
+		bool TooCloseToBuilding(CPos cell, ActorInfo actorInfo, BuildingInfo buildingInfo, BaseBuilderBotModule baseBuilder)
+		{
+			if (baseBuilder.Info.BaseSpacing == null)
+				return false;
+
+			if (!baseBuilder.Info.BaseSpacing.TryGetValue(actorInfo.Name, out var baseSpacing))
+				return false;
+
+			foreach (var tile in buildingInfo.Tiles(cell))
+			{
+				foreach (var adjacent in CVec.Directions)
+				{
+					var adjacentTile = tile + adjacent * baseSpacing;
```

Comment:
This should check all tiles from `tile + adjacent` to `tile + adjacent * baseSpacing`, shouldn't it? At the moment you can define a base spacing of e.g. 5 and the AI will build next to a 2x2 building which is "skipped over" by checking 5 tiles away here. Example with a spacing of 5 on every building:
![grafik](https://user-images.githubusercontent.com/7704140/153305075-cbc3dcc3-c7db-4369-a61c-58ea36927c04.png)

This alone wouldn't be that big of a problem (nobody is setting a spacing of 5 for every building), but you can't reliably have one building with a large spacing in the current code. That is because only the building that is placed is checking the spacing, but the spacing of already existing buildings is ignored. Is that intentional? (It seems counter-intuitive. You can set refineries to always have a spacing of 2, and power plants to have a spacing of 0. That might then result in a power plant being placed right next to the refinery. So you would have to set the spacing of power plants too, removing the possibility of packing them together...)

Reply:
Added a check for the whole distance of adjacent buildings.